### DISCUSSION TITLE
How to bind a datatable to a datagrid

### DISCUSSION THREAD

**Discussion by qq908382818** [AvaloniaUI/Avalonia#15046] [General]
DataTable dt1 = new();
dt1.Columns.Add("row1", typeof(string));
dt1.Columns.Add("row2", typeof(string));
for (int i = 0; i < 10; i++)
{
    DataRow dr1 = dt1.NewRow();
    dr1["row1"] = "row1_" + getRand();
    dr1["row2"] = "row2_" + getRand();
    dt1.Rows.Add(dr1);
}
this.dg1.ItemsSource = dt1.DefaultView;
DataGridTextColumn colTmp = new()
{
    Header = $"col{i}",//it's ok
    Binding = new Binding($"Row.ItemArray[{i}]"),
    CanUserSort = true,
    CanUserResize = true,
};

this.dg1.Columns.Add(colTmp);
use c# code is ok.how to bind to xamal

<--DataGrid x:Name="dg1" AutoGenerateColumns="False" 
		  >
	<--DataGrid.Columns>
		<--DataGridTextColumn Header="Row 1" Binding="{Binding [row1]}"/>//this is error 
	<--/DataGrid.Columns>
<--/DataGrid>

how to write axaml binding code
avaloniaui v11.0.10
 Cannot parse a compiled binding without an explicit x:DataType directive to give a starting data type for bindings. Line 19, position 41.	AvaloniaApplication3	F:\Project\111\AvaloniaApplication3\AvaloniaApplication3\MainWindow.axaml	19

**Comment by thevortexcloud** [comment]
Is there a specific reason you need a data table? They honestly just make things far more complicated in most instances.

Anyway, that specific error you are getting is because you have compiled bindings enabled. Either specify the data type or use a `ReflectionBinding` instead. Do note that reflection bindings may not work if you are reliant on AOT.

**Reply by qq908382818** [reply]
How to achieve this method through simple operations?
